pkg/artifactory/resource_artifactory_remote_repository.go

```diff
@@ -290,6 +314,14 @@ func unpackRemoteRepo(s *schema.ResourceData) *v1.RemoteRepository {
 	repo.VcsGitProvider = d.getStringRef("vcs_git_provider")
 	repo.VcsType = d.getStringRef("vcs_type")
 	repo.XrayIndex = d.getBoolRef("xray_index")
+	if v, ok := d.GetOk("nuget"); ok {
+		nugetConfig := v.(*schema.Set).List()[0].(map[string]interface{})
```

Comment:
If we convert the schema to typeList we don't need to cast via set and the indexing on zero can be done safely